protocol/storage/miner.go

```diff
@@ -114,6 +121,7 @@ func NewMiner(minerAddr, ownerAddr address.Address, workerAddr address.Address,
 		node:                nd,
 		proposalAcceptor:    acceptProposal,
 		proposalRejector:    rejectProposal,
+		storageFaultSlasher: slasher,
```

Comment:
Non-blocking: I have a feeling this should be operating in a separate event loop than the storage miner's.  Slashing other miners is as separate from sealing/posting as block mining is.  It would make sense for the slasher to be run in its own small "protocol" along with storage, retrieval, hello and block mining.  Fine for now.

Reply:
@acruikshank

Reply:
@shannonwells please document this reasoning here. I would in fact advocate for moving it out anyway, as this is unwelcome coupling between two processes that have only a slight interaction - it should at least be possible (easy) to mine *without* running a slashing monitor, depending on node configuration. One smell here is all the unnecessary diffs related to passing a `TrivialTestSlasher`everywhere that a miner is constructed.

Edit: and the miner test below! Please move it out in a follow-up.